api/app/controllers/api/v1/admin/listings_controller.rb

```
@@ -0,0 +1,197 @@
+            if tenant.nil?
+              render json: { errors: ['Nenhum tenant encontrado'] }, status: :unprocessable_entity
+              return
+            end
+
+            if tenant.scraper_source_url.blank?
+              render json: { errors: ['Este tenant não tem scraper configurado'] }, status: :unprocessable_entity
+              return
+            end
+
+            # Validate URL starts with tenant's scraper source
+            scraper_domain = URI.parse(tenant.scraper_source_url).host rescue nil
```

RuboCop (1.81.1)

[convention] Avoid using `rescue` in its modifier form.

(Style/RescueModifier)